features/vaadin-surveillance-views/src/test/java/org/opennms/features/vaadin/surveillanceviews/service/NMS15448_IT.java

```diff
@@ -0,0 +1,156 @@
+        alarm.setCounter(1);
+        alarm.setIpAddr(InetAddressUtils.getInetAddress("192.168.1.1"));
+        alarm.setSeverity(OnmsSeverity.NORMAL);
+        alarm.setFirstEventTime(event.getEventTime());
+        alarm.setLastEvent(event);
+        alarm.setServiceType(databasePopulator.getServiceTypeDao().findByName("ICMP"));
+        if (acknowledged) {
+            alarm.setAlarmAckUser("foobar");
+            alarm.setAlarmAckTime(new Date());
+        }
+        return alarm;
+    }
+
+    @Test
+    public void testSomething() {
+        final DefaultSurveillanceViewService surveillanceViewService = new DefaultSurveillanceViewService();
+        surveillanceViewService.setTransactionOperations(transactionOperations);
+        surveillanceViewService.setMonitoredServiceDao(databasePopulator.getMonitoredServiceDao());
+        surveillanceViewService.setOutageDao(databasePopulator.getOutageDao());
+        surveillanceViewService.setAlarmDao(databasePopulator.getAlarmDao());
+        final OnmsCategory cat1 = databasePopulator.getCategoryDao().findByName("Routers");
+        final OnmsCategory cat2 = databasePopulator.getCategoryDao().findByName("DEV_AC");
+
+        final List<OnmsAlarm> alarmsBefore = surveillanceViewService.getAlarmsForCategories(Sets.newHashSet(cat1), Sets.newHashSet(cat2));
+
+        Assert.assertEquals(1, alarmsBefore.size());
+
+        final OnmsEvent event1 = databasePopulator.buildEvent(distPoller);
+        event1.setEventCreateTime(new Date(1436881548292L));
+        event1.setEventTime(new Date(1436881548292L));
+        databasePopulator.getEventDao().save(event1);
+        databasePopulator.getEventDao().flush();
+
+        final OnmsAlarm alarm1 = buildAlarm(event1, false);
+        databasePopulator.getAlarmDao().save(alarm1);
+        databasePopulator.getAlarmDao().flush();
+
+        final OnmsEvent event2 = databasePopulator.buildEvent(distPoller);
+        event2.setEventCreateTime(new Date(1436881548292L));
+        event2.setEventTime(new Date(1436881548292L));
+        databasePopulator.getEventDao().save(event2);
+        databasePopulator.getEventDao().flush();
+
+        final OnmsAlarm alarm2 = buildAlarm(event2, true);
+        databasePopulator.getAlarmDao().save(alarm2);
+        databasePopulator.getAlarmDao().flush();
+
+        final OnmsEvent event3 = databasePopulator.buildEvent(distPoller);
+        event3.setEventCreateTime(new Date(1436881548292L));
+        event3.setEventTime(new Date(1436881548292L));
+        databasePopulator.getEventDao().save(event3);
+        databasePopulator.getEventDao().flush();
+
+        final OnmsAlarm alarm3 = buildAlarm(event3, false);
+        databasePopulator.getAlarmDao().save(alarm3);
+        databasePopulator.getAlarmDao().flush();
+
+        final List<OnmsAlarm> alarmsAfter = surveillanceViewService.getAlarmsForCategories(Sets.newHashSet(cat1), Sets.newHashSet(cat2));
+
+        Assert.assertEquals(3, alarmsAfter.size());
```

Comment:
Consider adding a comment (either inline or for the method) about this result, e.g. "There should be 3 unacknowledged alarms: the initial one from the databasePopulator, plus 2 of the 3 added in the test"

Reply:
yeah, we should consider it a code smell to use an assert that doesn't use the 3-argument version (or 2, for `assertTrue`/`assertFalse`)

Reply:
done